_data-prepper/pipelines/configuration/processors/list-to-map.md

```diff
@@ -16,10 +16,12 @@ The following table describes the configuration options used to generate target
 
 Option | Required | Type | Description
 :--- | :--- | :--- | :---
-`key` | Yes | String | The key of the fields to be extracted as keys in the generated mappings.
 `source` | Yes | String | The list of objects with `key` fields to be converted into keys for the generated map.
 `target` | No | String | The target for the generated map. When not specified, the generated map will be placed in the root node.
+`key` | Conditionally | String | The key of the fields to be extracted as keys in the generated mappings. Must be specified if `use_source_key` is `false`.
+`use_source_key` | No | Boolean | When `true`, keys in the generated map will use original keys from the source. Default is `false`.
 `value_key` | No | String | When specified, values given a `value_key` in objects contained in the source list will be extracted and converted into the value specified by this option based on the generated map. When not specified, objects contained in the source list retain their original value when mapped.
+`extract_value` | No | Boolean | When `true`, values in objects from the source list will be extracted and put into the generated map. When `false`, objects from the source list are put into the generated map with their values listed the same as they were found in the source list. Default is `false`
```

Comment:
"object values" instead of "values in objects"? I'm not following "with their values listed the same as they were found in the source list". Do we mean something like "with their values unchanged"?

Reply:
The object values don't change in either context, rather, the way in which the values appear in the generated map. When "extract_value" is set to `false`, the object values appear in the generated map as they were presented in the source, for in a source name `myList`:

```
"mylist": [
    {
      "name": "a",
      "value": "val-a"
    },
```

Would appear as in the generated map: 

```
{
  "name": ["a", "b"],
  "value": [
      {
        "name": "a",
        "value": "val-a"
      },
```

When extract values is set to `true`, the values are extracted from their original objects and in the same line in the generated map:

```
{
  "name": ["a", "b"],
  "value": ["val-a", "val-b"]
}
```